src/gino/loader.py

```diff
@@ -117,7 +117,13 @@ def do_load(self, row, context):
         else:
             for key, value in self.extras.items():
                 value, distinct_ = value.do_load(row, context)
-                if distinct_ is not None:
+                if distinct_ is None:
+                    continue
+                if hasattr(self.model, key) and not isinstance(
+                    getattr(self.model, key), property
+                ):
```

Comment:
~(trivial) can be simplified as:~

```python
if not isinstance(getattr(self.model, key, None), (property, type(None))):
```
---
I'm so sorry - while writing above comment, I thought about another scenario that we may break here:

```python
class Book(db.Model):
    __tablename__ = "authors"

    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.Unicode)
    author = None
```

And:

```python
Book.load(author=Author)
```

We may need to check for function type?

```python
if isinstance(getattr(self.model, key, None), types.FunctionType):
```

Reply:
Yes, agree.
Added check for `types.FunctionType`